sebs/aws/aws.py

```diff
@@ -157,7 +166,145 @@ def package_code(
         mbytes = bytes_size / 1024.0 / 1024.0
         self.logging.info("Zip archive size {:2f} MB".format(mbytes))
 
-        return os.path.join(directory, "{}.zip".format(benchmark)), bytes_size
+        return os.path.join(directory, "{}.zip".format(benchmark)), bytes_size, container_uri
+
+    # PK: To Do: Test
+    def find_image(self, repository_client, repository_name, image_tag) -> bool:
+        try:
+            response = repository_client.describe_images(
+                repositoryName=repository_name,
+                imageIds=[{'imageTag': image_tag}]
+            )
+            if response['imageDetails']:
+                return True
+        except ClientError as e:
+            return False
+
+    def push_image_to_repository(self, repository_client, repository_uri, image_tag):
+        auth = repository_client.get_authorization_token()
+        auth_data = auth['authorizationData'][0]
+        token = base64.b64decode(auth_data['authorizationToken']).decode('utf-8')
+        username, password = token.split(':')
+        registry_url = auth_data['proxyEndpoint']
+
+        self.docker_client.login(username=username, password=password, registry=registry_url)
```

Comment:
I would encapsulate authorization in a separate function.

Reply:
Shouldn't this also be aware of config: if the user provides a registry and credentials in the config, we use that one?

Reply:
Yessss, 
I changed the logic so now, the docker client is logged in the constructor ( like in openwhisk ) if the user has provided the credentials, else we just log in by getting the username, and password from ECR. 

Docker login (https://github.com/moby/moby/issues/15466) does not save state so it is not possible to later check if it is already logged in or not unless we query  ~/.docker/config.json which I believe is not a good idea, or we could use some global flag but I reckon just doing try except will be ideal here. 

let me know what you think

Reply:
Not quite sure if try .. except is a good choice here
Please let me know if you have any suggestions